overlord/snapstate/snapstate.go

```diff
@@ -2000,6 +2003,36 @@ func autoRefreshPhase1(ctx context.Context, st *state.State) ([]string, []*state
 		}
 	}
 
+	// if specific gating snap was given, drop other affected snaps unless
+	// they are affected by same updates as forGatingSnap.
+	if forGatingSnap != "" {
+		if _, ok := affectedSnaps[forGatingSnap]; !ok {
+			// this can happen if after refreshing candidates the snap is no
+			// longer affected by new candidates.
```

Comment:
this is a bit confusing to me. Being affected doesn't really change whether there's an update for it or not. We should run an update for it even if nothing is affecting it. The reality is that usually the snap will be added to affectedSnaps because of its own hook but again not sure we should base our decision to do something here on that. The easiest condition to understand is whether forGatingSnap is in `updated` or not at all.

Reply:
Fair point, tweaked.